crates/harness/src/monolith.rs

```diff
@@ -58,6 +62,13 @@ pub(crate) struct MonolithState {
 
 impl Monolith {
     pub async fn new(ctx: &TestRunner) -> anyhow::Result<Self> {
+        Self::with_behavior(ctx, Box::new(BehaviorManual)).await
+    }
+
+    pub async fn with_behavior(
+        ctx: &TestRunner,
+        behavior: Box<dyn Behavior + Send + 'static>,
```

Comment:
I looked at the docs for dyn, I'm kinda confused. Why does this need to be dynamically dispatched?

Reply:
This avoids needing to have Monolith be a generic, which was causing other problems for me. I don't remember the specific reasons I did it like this.